modules/browsiRtdProvider.js

```diff
@@ -157,36 +167,74 @@ function getKVObject(p, keyName) {
 /**
  * check if placement id matches one of given ad units
  * @param {number} id placement id
- * @param {Object[]} allSlots google slots on page
+ * @param {Object} slot google slot
  * @param {string[]} whitelist ad units
  * @return {boolean}
  */
-export function isIdMatchingAdUnit(id, allSlots, whitelist) {
+export function isIdMatchingAdUnit(id, slot, whitelist) {
   if (!whitelist || !whitelist.length) {
     return true;
   }
-  const slot = allSlots.filter(s => s.getSlotElementId() === id);
-  const slotAdUnits = slot.map(s => s.getAdUnitPath());
-  return slotAdUnits.some(a => whitelist.indexOf(a) !== -1);
+  const slotAdUnits = slot.getAdUnitPath();
+  return whitelist.indexOf(slotAdUnits) !== -1;
+}
+
+/**
+ * get GPT slot by placement id
+ * @param {string} id placement id
+ * @return {?Object}
+ */
+function getSlotById(id) {
+  const slots = getAllSlots();
+  if (!slots || !slots.length) {
+    return null;
+  }
+  return slots.filter(s => s.getSlotElementId() === id)[0] || null;
+}
+
+/**
+ * generate id according to macro script
+ * @param {string} macro replacement macro
+ * @param {string} id placement id
+ * @param {Object} slot google slot
+ * @return {?Object}
+ */
+function getMacroId(macro, id, slot) {
+  if (macro) {
+    try {
+      const macroString = macro
+        .replace(/<DIV_ID>/g, `${id}`)
+        .replace(/<AD_UNIT>/g, `${slot.getAdUnitPath()}`)
+        .replace(/<KEY_(\w+)>/g, (match, p1) => {
+          return (p1 && slot.getTargeting(p1).join('_')) || 'NA';
+        });
+      return eval(macroString);// eslint-disable-line no-eval
+    } catch (e) {
+      utils.logError(`failed to evaluate: ${macro}`);
+    }
+  }
+  return id;
 }
 
 /**
  * XMLHttpRequest to get data form browsi server
  * @param {string} url server url with query params
  */
 function getPredictionsFromServer(url) {
+  let ajax = ajaxBuilder(_moduleParams.auctionDelay || _moduleParams.timeout || DEF_TIMEOUT);
```

Comment:
I think `auctionDelay` should be up at the higher level underneath realTimeData, not within the dataProviders section.

Reply:
It is. `auctionDelay` and `timeout` are both underneath realTimeData
I push those values to `_moduleParams ` on init to keep all relevant configs for the module on `_moduleParams `